packit_service/worker/build/build_helper.py

```diff
@@ -373,9 +411,59 @@ def _report(
         The status reporting should be done through this method
         so we can extend it in subclasses easily.
         """
-        self.status_reporter.report(
-            description=description, state=state, url=url, check_names=check_names,
-        )
+        if (
+            isinstance(self.project, GitlabProject)
+            and not self.check_project_permission
+        ):
+            # "Packit-User" does not have access to project.
+            self.request_project_access()
+            self.report_status_by_comment(state, url, check_names)
+        else:
+            self.status_reporter.report(
+                description=description, state=state, url=url, check_names=check_names,
+            )
+
+    def report_status_by_comment(
+        self, state: CommitStatus, url: str, check_names: Union[str, list, None],
+    ) -> None:
+        """
+        Reporting build status with MR comment if no permission to the fork project
+        """
+        comment_msg = f"[{check_names}]({url}): {state.name.upper()}"
+
+        all_pr_comments = self.project.get_pr(pr_id=self.pr_id)._get_all_comments()[
+            ::-1
+        ]
+
```

Comment:
Don't we have some argument to do this with the public method?

Reply:
Thanks! Found it.